src/layers/conv.jl

```diff
@@ -25,127 +26,59 @@ Create a convolution + batch normalization pair with activation.
   - `pad`: padding of the convolution kernel
   - `dilation`: dilation of the convolution kernel
   - `groups`: groups for the convolution kernel
-  - `bias`, `weight`, `init`: initialization for the convolution kernel (see [`Flux.Conv`](#))
+  - `bias`: bias for the convolution kernel. This is set to `false` by default if
+    `use_norm = true`.
+  - `weight`, `init`: initialization for the convolution kernel (see [`Flux.Conv`](#))
 """
-function conv_norm(kernel_size, inplanes::Integer, outplanes::Integer, activation = relu;
-                   norm_layer = BatchNorm, revnorm::Bool = false, preact::Bool = false,
-                   use_norm::Bool = true, kwargs...)
+function conv_norm(kernel_size::Dims{2}, inplanes::Integer, outplanes::Integer,
+                   activation = relu; norm_layer = BatchNorm, revnorm::Bool = false,
+                   eps::Float32 = 1.0f-5, preact::Bool = false, use_norm::Bool = true,
+                   momentum::Union{Nothing, Number} = nothing, bias = !use_norm, kwargs...)
+    # no normalization layer
     if !use_norm
-        if (preact || revnorm)
+        if preact || revnorm
             throw(ArgumentError("`preact` only supported with `use_norm = true`"))
         else
+            # early return if no norm layer is required
             return [Conv(kernel_size, inplanes => outplanes, activation; kwargs...)]
         end
     end
+    # channels for norm layer and activation functions for both conv and norm
     if revnorm
         activations = (conv = activation, bn = identity)
-        bnplanes = inplanes
+        normplanes = inplanes
     else
         activations = (conv = identity, bn = activation)
-        bnplanes = outplanes
+        normplanes = outplanes
     end
+    # handle pre-activation
     if preact
         if revnorm
             throw(ArgumentError("`preact` and `revnorm` cannot be set at the same time"))
         else
             activations = (conv = activation, bn = identity)
         end
     end
-    layers = [Conv(kernel_size, inplanes => outplanes, activations.conv; kwargs...),
-        norm_layer(bnplanes, activations.bn)]
+    # handle momentum for BatchNorm
+    if !isnothing(momentum)
+        @assert norm_layer==BatchNorm "`momentum` is only supported for `BatchNorm`"
+        norm_layer = (args...; kwargs...) -> BatchNorm(args...; momentum, kwargs...)
+    end
```

Comment:
Originally, we had the `BatchNorm` arguments specifically listed. Then with `norm_layer`, we removed them. The reason I liked that change is that it avoids `kwargs...` passing. Meaning that if I want a `BatchNorm` with a specific momentum, then the caller should be doing `conv_norm(...; norm_layer = (args...; kwargs...) -> BatchNorm(args...; momentum = MY_NUM, kwargs...)`. Then you can also remove all the `momentum` / `nothing` handling from `layers/mbconv.jl`.

Right now, we are just special casing one argument, so it seems okay. But this type of design scales poorly as we need more arguments. The closure based one via `norm_layer` scales well to arbitrary norm layers and their arguments. Instead, `mnasnet` should specify `norm_layer` as above.

Reply:
Yeah this makes sense - we might just need to redirect users to a section on using closures from the Julia docs when we write docs for this so that they're comfortable with the syntax needed

Reply:
Yes, but also the revamped docs should have a "how-to" guide on `conv_norm` and friends. Directly spell out this use case then only relying on the linked Julia docs + our docstring.

Reply:
Resolved, but leaving this here so that I remember to add docs for this